src/main/java/de/tum/in/www1/artemis/web/rest/theia/TheiaConfigurationResource.java

```diff
@@ -0,0 +1,43 @@
+package de.tum.in.www1.artemis.web.rest.theia;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_THEIA;
+
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.config.TheiaConfiguration;
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastInstructor;
+
+@Profile(PROFILE_THEIA)
+@RestController
+@RequestMapping("api/theia/")
+public class TheiaConfigurationResource {
+
+    private final TheiaConfiguration theiaConfiguration;
+
+    @Autowired
```

Comment:
we do not want to use @Autowired (it's only allowed in exceptional cases) and it's not necessary for constructor injection